Makefile

```diff
@@ -104,6 +104,7 @@ test-unit: ## Run unit tests
 
 test-acc: dev-up ## Run acceptance tests
 	${call print, "Running acceptance tests"}
+	@docker-compose exec -- auth0 curl -4 -sS -o /dev/null -X POST http://localhost:8080/__admin/scenarios/reset
```

Comment:
This fixes a test bug where running the acceptance tests multiple times in a row required a `dev-down` and `dev-up` cycle.

Reply:
First of all, I'd like to really commend the great work you've put into this PR and the follow up to make sure our testing is thorough. 🥳  You've really went above and beyond and we can't thank you enough. 

I generally like the improvements a lot however I encounter a few hiccups while reviewing and testing this locally. Let me explain:

- Everything works great if I always make sure to run all the tests through the `make test-acc` command
- However if I only run `TestAccCustomDomainVerificationWithSelfManagedCerts` or `TestAccCustomDomainVerificationWithAuth0ManagedCerts` at the second consecutive run I get `Error: invalid character 'R' looking for beginning of value` because now we're required to reset the scenarios before every run. So I have to make sure not to forget to do `docker-compose exec -- auth0 curl -4 -sS -o /dev/null -X POST http://localhost:8080/__admin/scenarios/reset`
- Before these changes I ran the previous test we had in a loop 100 times and it never failed without resetting
-  So I'm wondering what can we do in order for us to have the same behavior as before and not need to reset the scenarios after each test as it gets to be quite a cumbersome task to do?

Reply:
So, interestingly enough, previously this worked only coincidentally. The attribute checks in the test were valid against both the "Started" and "Verified" responses. The first time you ran the test, it would correctly do the scenario state transition from the "Started" state, and then after that every test would just run against the "Verified" responses from the outset. You can validate that by checking the `status` attribute in the test.

I changed the transitions so that they have to follow the create-to-verify path, which is what's breaking now.

I was using `make test-acc TESTARGS="-run '.*TestAccCustomDomainVerification.*'"`, but I didn't test with `go test` directly. However that use case makes sense, and I think what we can actually do there is have each test reset the specific scenarios it cares about before running. Let me do a little work on that and get back to you!

Reply:
The feature on the WireMock side to reset specific scenarios is new and their Docker image is broken or hasn't been updated yet (not sure what's going on there), so I opted to create a little client that the tests will simply call once to reset all mappings/scenarios the first time a mock provider is instantiated. That should give parity between calling `go test` in the Makefile and standalone. Let me know what you think!

Reply:
You're totally right @impl, we weren't checking for the status attribute and indeed before a subsequent call would make the test fail with that assertion present. Really appreciate you spotting that and fixing it! 🙇🏻 

I reviewed the PR and really like the approach 👏🏻 , as I wanted to save you some time I went ahead and refactored slightly the wiremock client package to remove the 3rd party retry sdk as IMO it's unnecessary. As I couldn't push on top of this PR as you had the "enable maintainers to modify contribution" checkbox thingie disabled I forked your fork and appended the changes over there https://github.com/puppetlabs/terraform-provider-auth0/pull/1. Let me know as well what you think:) and we can merge that into this PR or you can enable the checkbox on this PR so I can push on top, up to you. 

Thanks a lot for all the hard work put into this! It's a really great contribution.